exercise_ashleeph.Rmd

```diff
@@ -0,0 +1,92 @@
+
+8.  **Review your assigned pull request**. You should receive a notification either via GitHub (you can check by clicking the 'notifications' icon directly next to your profile photo in the upper right corner), or you will receive an email link to the email you registered your GitHub account to.
+
+    Follow [these guidelines for reviewing a pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/reviewing-proposed-changes-in-a-pull-request), give some meaningful feedback, and when you're done 'Approve' the pull request (do NOT close or merge).
+
+## The coding exercise:
+
+Work through the steps below to complete this exercise (to the best of your ability) and then save, add, commit and push this file with your coding edits to your forked repo and follow the instructions above for completing the assignment via creating and reviewing a pull request.
+
+Load in the tidyverse (only package(s) needed for this exercise:
+
+```{r}
+library(tidyverse)
+library(dplyr)
+```
+
+Read in the 'kenya_county_climates.csv' in the 'data/' folder and assign it to an environmental variable (Hint: use 'read_csv()').
+
+```{r}
+Kenya <- read.csv("data/kenya_county_climates.csv")
+```
+
+Write a chain of operations that does the following:
+
+1.  `select()` just the columns that have "Precipitation" in the name (so we are only interested in the precipitation data), AND the 'County' column.
+```{r}
+KenyaPre <- Kenya %>%
+select(County,contains("Precipitation"))
+```
+
+
+2.  `filter()` all the rows where 'Precipitation_585_90' is **greater than** the mean of all county precipitation values.
+```{r}
+KenyaPre <- Kenya %>%
+select(County,contains("Precipitation")) %>%
+filter(Precipitation_585_90 > mean(Precipitation_585_90,na.rm= F))
+```
+
+
+3.  `pivot_longer()` all the precipitation columns (so everything *except* 'County'), so we have one column for all precip values and one column with the value's associated variables.
+
+```{r}
+  Kenyapivot <- pivot_longer(KenyaPre,contains("Precipitation"))
+
+```
+
+Question: How many counties have predicted precip values greater than the total average for all counties?
+Using the new dataset created above, create a bar plot of precipitation for each time frame for each county using `facet_wrap()`
+
+```{r}
+Kenyapivot%>%
+  ggplot() +
+  geom_col(aes(x=name, y=value)) +
+  facet_wrap(county)
+```
+
+CHALLENGE:
+
+This is messy though since we have so many counties. Instead, use `map()` to create an individual bar plot for each County (hint: create a vector of county names from the data, and building off the ggplot code you made above, add in a line to `filter()` out each county (i.e., .x) before creating the plot. AND, no more need to `facet_wrap()` here).
+
+There are many ways to do this!
+
```

Comment:
I also was not able to get to the challenge question, but here is the solution:

counties <- unique(data_filtered$County)

plots <-
  map(counties,
      ~ data_filtered %>% filter(County == .x) %>% ggplot(aes(x = name, y = value)) + geom_col())